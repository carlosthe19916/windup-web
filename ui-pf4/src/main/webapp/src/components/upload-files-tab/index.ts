export { UploadFilesTab } from "./upload-files-tab";
