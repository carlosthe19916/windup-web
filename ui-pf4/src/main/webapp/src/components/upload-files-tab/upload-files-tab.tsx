import * as React from "react";
import { useState } from "react";
import { AxiosPromise, AxiosError } from "axios";
import { Tabs, Tab, TabTitleText } from "@patternfly/react-core";
import { UploadFilesSection } from "../upload-files-section";
import { Application } from "../../models/api";

export interface UploadFilesTabProps {
  fileFormName: string;
  applications?: Application[];
  uploadFile: (formData: FormData, config: any) => AxiosPromise;
  removeFile?: (file: File) => void;
  onUploadFileSuccess?: (response: any, file: File) => void;
  onUploadFileError?: (error: AxiosError, file: File) => void;
}

export const UploadFilesTab: React.FC<UploadFilesTabProps> = ({
  fileFormName,
  uploadFile,
  removeFile,
  onUploadFileSuccess,
  onUploadFileError,
}) => {
  const [activeTabKey, setActiveTabKey] = useState<number | string>(0);

  const handleTabClick = (_: any, tabIndex: number | string) => {
    setActiveTabKey(tabIndex);
  };

  return (
    <Tabs activeKey={activeTabKey} onSelect={handleTabClick}>
      <Tab eventKey={0} title={<TabTitleText>Upload</TabTitleText>}>
        <br />
        <UploadFilesSection
          fileFormName={fileFormName}
          accept=".ear,.har,.jar,.rar,.sar,.war,.zip"
          upload={uploadFile}
          onRemove={removeFile}
          onSuccess={(response, file: File) => {
            if (onUploadFileSuccess) {
              onUploadFileSuccess(response, file);
            }
          }}
          onError={(error, file) => {
            if (onUploadFileError) {
              onUploadFileError(error, file);
            }
          }}
        />
      </Tab>
      <Tab eventKey={1} title={<TabTitleText>Directory path</TabTitleText>}>
        <p>Not implemented yet</p>
      </Tab>
    </Tabs>
  );
};
