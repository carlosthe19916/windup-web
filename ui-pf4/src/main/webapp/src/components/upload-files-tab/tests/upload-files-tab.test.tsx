import * as React from "react";
import { shallow } from "enzyme";
import { UploadFilesTab } from "../upload-files-tab";

describe("Welcome", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<UploadFilesTab />);
    expect(wrapper).toMatchSnapshot();
  });
});
