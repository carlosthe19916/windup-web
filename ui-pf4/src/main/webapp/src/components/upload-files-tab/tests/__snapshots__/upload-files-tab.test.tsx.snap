// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Renders without crashing 1`] = `
<Tabs
  activeKey={0}
  component="div"
  isBox={false}
  isFilled={false}
  isSecondary={false}
  isVertical={false}
  leftScrollAriaLabel="Scroll left"
  mountOnEnter={false}
  onSelect={[Function]}
  ouiaSafe={true}
  rightScrollAriaLabel="Scroll right"
  unmountOnExit={false}
>
  <Tab
    eventKey={0}
    title={
      <TabTitleText>
        Upload
      </TabTitleText>
    }
  >
    <br />
    <UploadFilesSection
      onError={[Function]}
      onSuccess={[Function]}
    />
  </Tab>
  <Tab
    eventKey={1}
    title={
      <TabTitleText>
        Directory path
      </TabTitleText>
    }
  >
    <p>
      Not implemented yet
    </p>
  </Tab>
</Tabs>
`;
